### PR TITLE
Remove Kokkos::vector

### LINKED ISSUE #1019: Kokkos::vector'is deprecated
I started to see this warning since I upgraded Trilinos:
```
In file included from /home/lbertag/workdir/albany/albany-src/branch/src/evaluators/gather/PHAL_GatherSolution.cpp:9:
/home/lbertag/workdir/albany/albany-src/branch/src/evaluators/gather/PHAL_GatherSolution.hpp:135:24: warning: 'template<class Scalar, class Arg1Type> class Kokkos::vector' is deprecated [-Wdeprecated-declarations]
  135 |     using KV = Kokkos::vector<DynRankView, PHX::Device>;
      |                        ^~~~~~
In file included from /home/lbertag/workdir/albany/albany-src/branch/src/evaluators/gather/PHAL_GatherSolution.hpp:23,
                 from /home/lbertag/workdir/albany/albany-src/branch/src/evaluators/gather/PHAL_GatherSolution.cpp:9:
/storage/workdir/trilinos/trilinos-install/groppello/gcc/opt/develop-serial/include/Kokkos_Vector.hpp:52:25: note: declared here
   52 | class KOKKOS_DEPRECATED vector
```
While I don't expect the class to disappear very soon, we should be proactive, and remove its usage. We don't want to forget and have to rush it when Kokkos finally removes it.

### PR DESCRIPTION
**Pull request by jewatkins**
Fixes #1019

works with clang, currently testing on weaver

### CODE REVIEW

**Comment by jewatkins** [comment]
weaver tests pass.

File: src/evaluators/gather/PHAL_GatherSolution_Def.hpp [hunk 1]
```diff
@@ -172,20 +172,23 @@ postRegistrationSetup(typename Traits::SetupData d,
 
     // Get MDField views from std::vector
```
Comment (mperego) [inline src/evaluators/gather/PHAL_GatherSolution_Def.hpp:173]: This comment seems wrong

File: src/evaluators/scatter/PHAL_ScatterResidual_Def.hpp [hunk 2]
```diff
@@ -101,9 +101,10 @@ postRegistrationSetup(typename Traits::SetupData d,
 
     for (int eq =0; eq<numFields;eq++){
       // Get MDField views from std::vector
```
Comment (mperego) [inline src/evaluators/scatter/PHAL_ScatterResidual_Def.hpp:103]: This comment seems wrong

**Review by mperego — APPROVED** [review]
Looks good to me. Thanks!

**Comment by mperego** [comment]
Thanks!

**Pull request merged**